janitor/functions/select.py

```diff
@@ -27,21 +27,180 @@ def select_columns(
+           brainwt   bodywt  conservation
+        0      NaN   50.000            lc
+        1  0.01550    0.480           NaN
+        2      NaN    1.350            nt
+        3  0.00029    0.019            lc
+        4  0.42300  600.000  domesticated
+
+        Selection on MultiIndex columns:
+        >>> d = {'num_legs': [4, 4, 2, 2],
+        ...      'num_wings': [0, 0, 2, 2],
+        ...      'class': ['mammal', 'mammal', 'mammal', 'bird'],
+        ...      'animal': ['cat', 'dog', 'bat', 'penguin'],
+        ...      'locomotion': ['walks', 'walks', 'flies', 'walks']}
+        >>> df = pd.DataFrame(data=d)
+        >>> df = df.set_index(['class', 'animal', 'locomotion']).T
         >>> df
-           col1  foo  col2
-        0     1    3     5
-        1     2    4     6
-        >>> df.select_columns("col*")
-           col1  col2
-        0     1     5
-        1     2     6
+        class      mammal                bird
+        animal        cat   dog   bat penguin
+        locomotion  walks walks flies   walks
+        num_legs        4     4     2       2
+        num_wings       0     0     2       2
+
+        Selection with a scalar:
+        >>> df.select_columns('mammal')
+        class      mammal
+        animal        cat   dog   bat
+        locomotion  walks walks flies
+        num_legs        4     4     2
+        num_wings       0     0     2
+
+        Selection with a tuple:
+        >>> df.select_columns(('mammal','bat'))
+        class      mammal
+        animal        bat
+        locomotion  flies
+        num_legs        2
+        num_wings       2
+
+        Selection within a level is possible with a dictionary,
+        where the key is either a level name or number:
+        >>> df.select_columns({'animal':'cat'})
+        class      mammal
+        animal        cat
+        locomotion  walks
+        num_legs        4
+        num_wings       0
+        >>> df.select_columns({1:["bat", "cat"]})
+        class      mammal
+        animal        bat   cat
+        locomotion  flies walks
+        num_legs        2     4
+        num_wings       2     0
+
+        Selection on multiple levels:
+        >>> df.select_columns({"class":"mammal", "locomotion":"flies"})
+        class      mammal
+        animal        bat
+        locomotion  flies
+        num_legs        2
+        num_wings       2
+
+        Selection with a regex on a level:
+        >>> df.select_columns({"animal":re.compile(".+t$")})
+        class      mammal
+        animal        cat   bat
+        locomotion  walks flies
+        num_legs        4     2
+        num_wings       0     2
+
+        Selection with a callable on a level:
+        >>> df.select_columns({"animal":lambda f: f.str.endswith('t')})
+        class      mammal
+        animal        cat   bat
+        locomotion  walks flies
+        num_legs        4     2
+        num_wings       0     2
+
```

Comment:
nit: Just one empty newline between consecutive sections will suffice.

Reply:
@thatlittleboy which line are you referring to?

Reply:
@samukweku L203 (there are currently 2 new lines between the "Examples" section and the "Args" section, we just need one. :)